app/services/MeansOfTransportIdentificationTypesService.scala

```diff
@@ -65,6 +65,16 @@ class MeansOfTransportIdentificationTypesService @Inject() (
           )
     }
 
+  def getMeansOfTransportIdentificationType(userAnswersCode: Option[String])(implicit hc: HeaderCarrier): Future[Option[TransportMeansIdentification]] =
+    referenceDataConnector.getMeansOfTransportIdentificationTypes.map {
+      identity =>
+        userAnswersCode match {
+          case Some(value) => identity.find(_.code.head == value.head)
+          case _           => None
+        }
+
```

Comment:
We can pass this as a query parameter rather than handling the entire list in memory. See `getSupportingDocument` and `getTransportDocument` as an example